app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1187,6 +1193,54 @@ private void initThumbnail(final String url) {
                 .loadImage(url, ImageDisplayConstants.DISPLAY_THUMBNAIL_OPTIONS, this);
     }
 
+    public void updateEndScreenThumbnail() {
+        if (currentThumbnail == null) {
+            return;
+        }
+
+        final float endScreenHeight;
+
+        updateScreenSize(); // ensure that screenHeight is initialized and thus not 0
+
+        // Show at least stream title and content creator on TVs and tablets (using the land layout)
+        // 85dp for height of R.id.detail_root and R.id.detail_title_root_layout
+        // 15/16sp for text size of video title
+        if (DeviceUtils.isTv(context) && !isFullscreen) {
+            final int videoInfoHeight =
+                    DeviceUtils.dpToPx(85, context) + DeviceUtils.spToPx(16, context);
+            endScreenHeight = Math.min(currentThumbnail.getHeight(),
+                    screenHeight - videoInfoHeight);
+        } else if (DeviceUtils.isTablet(context) && !isFullscreen) {
+            final int videoInfoHeight =
+                    DeviceUtils.dpToPx(85, context) + DeviceUtils.spToPx(15, context);
+            endScreenHeight = Math.min(currentThumbnail.getHeight(),
+                    screenHeight - videoInfoHeight);
+        } else { // fullscreen player: max height is the device height
+            endScreenHeight = Math.min(currentThumbnail.getHeight(), screenHeight);
+        }
```

Comment:
If you extract this into a method, you can 

1. Give it a name, which make this part here easier to read
2. Reference the `R.id.*` properly through javadoc instead of inline comments, which gives IDE support when refactoring or navigating. I don't think Android Studio will pick these up, if they are changed.

Something like `calculateScreenHeight`